ee/desktop/menu/menu.go

```diff
@@ -1,36 +1,174 @@
 package menu
 
 import (
-	"fmt"
+	"os/exec"
+	"runtime"
 
 	"fyne.io/systray"
-	"github.com/kolide/kit/version"
+
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
 	"github.com/kolide/launcher/ee/desktop/assets"
 )
 
-func Init(hostname string) {
-	onReady := func() {
-		systray.SetTemplateIcon(assets.KolideDesktopIcon, assets.KolideDesktopIcon)
-		systray.SetTooltip("Kolide")
+// MenuData encapsulates a systray icon and accessible menu items
+type MenuData struct {
+	Tooltip string         `json:"tooltip,omitempty"`
+	Items   []MenuItemData `json:"items"`
+}
 
-		versionItem := systray.AddMenuItem(fmt.Sprintf("Version %s", version.Version().Version), "")
-		versionItem.Disable()
+// MenuItemData represents a menu item, optionally containing sub menu items
+type MenuItemData struct {
+	Label       string         `json:"label,omitempty"`
+	Tooltip     string         `json:"tooltip,omitempty"`
+	Disabled    bool           `json:"disabled,omitempty"`
+	NonProdOnly bool           `json:"nonProdOnly,omitempty"`
+	IsSeparator bool           `json:"isSeparator,omitempty"`
+	Action      MenuItemAction `json:"action,omitempty"`
+	Items       []MenuItemData `json:"items,omitempty"`
+}
 
-		// if prod environment, return
-		if hostname == "k2device-preprod.kolide.com" || hostname == "k2device.kolide.com" {
-			return
-		}
+// MenuItemAction encapsulates what action should be performed when a menu item is invoked
+type MenuItemAction struct {
```

Comment:
Should this be a struct? Will we support things other than URLs? Is it just a type alias for string? (And if it's only going to be URLs, we should just call it URL, and hang it off `MenuItemData`)

Reply:
If it's meant to encasulate an action, an interface might be a better choice. Not sure that fits with systray though

Reply:
I'm envisioning there could be different types of actions - for now the only one I know we'll need is "open a browser to a URL". So I imagine this struct would become bigger, and allow for any one of these actions & their associated data to be encapsulated in it.

Some possible actions could be "Open a window", Debug/NonProd operations like "Restart the process", or "Refresh the menu", etc. maybe there's others folks have in mind I'm not aware of.

Reply:
If it's many possible actions it feels better represented by an interface. Like there's a `urlAction`, `windowAction`, etc,  and they each conform to a:
```
type menuItemAction interface {
  Performa(...)
  }
  ```

Reply:
I liked that idea! Now I have a data struct `actionData` which implements the `Action` interface. Definitely feels like a cleaner separation between the action invoker and the action performer code.